roles/infrared-horizon-selenium/tasks/main.yml

```diff
@@ -75,43 +75,53 @@
       include_tasks: configure_selenium_config.yml
       when: selenium_tests_temp.path is defined
 
-    - name: Remove old project and user
-      become: true
-      become_user: stack
-      shell: |
-        source /home/stack/overcloudrc
-        openstack project delete horizontest
-        openstack user delete horizontest
-      ignore_errors: yes
+    - name: Remove old project
+      os_project:
+        cloud: overcloud
+        state: absent
+        name: horizontest
+
+    - name: Remove old user
+      os_user:
+        cloud: overcloud
+        state: absent
+        name: horizontest
 
     - name: Create a project
-      become: true
-      become_user: stack
-      shell: |
-        source /home/stack/overcloudrc
-        openstack project create --description 'Horizon Selenium test project' horizontest
+      os_project:
+        cloud: overcloud
+        state: present
+        name: horizontest
+        description: Horizon Selenium test project
+        domain_id: default
+        enabled: True
 
     - name: Add horizontest user to project
-      become: true
-      become_user: stack
-      shell: |
-        source /home/stack/overcloudrc
-        openstack user create --project horizontest --password secretadmin horizontest
+      os_user:
+        cloud: overcloud
+        state: present
+        name: horizontest
+        password: secretadmin
+        domain: default
+        default_project: horizontest
 
     - name: Add horizontest user and horizontest project to member role
-      become: true
-      become_user: stack
-      shell: |
-        source /home/stack/overcloudrc
-        openstack role add --user horizontest --project horizontest member
-
-    - name: Add a flavor if it doesn't exist yet
-      become: true
-      become_user: stack
-      shell: |
-        source /home/stack/overcloudrc
-        openstack flavor create --public m1.tiny --ram 512 --disk 1 --vcpus 1 --rxtx-factor 1
-      ignore_errors: yes
+      os_user_role:
+        cloud: overcloud
+        user: horizontest
+        role: member
+        project: horizontest
+        
+    - name: Create flavor
```

Comment:
What happens if this flavor already exist?
Does the ansible module exhibit idempotency? or will it generate another flavor with a different ID?
If it is the latter, we may want to register the result and use that ID

Reply:
If the flavor already exists with the same name the ansible module deletes the previous one and generate a new one with provided name